parts/linux/cloud-init/artifacts/cse_main.sh

```diff
@@ -91,7 +91,15 @@ if [[ ${ID} != "mariner" ]]; then
     logs_to_events "AKS.CSE.removeManDbAutoUpdateFlagFile" removeManDbAutoUpdateFlagFile
 fi
 
-if [[ "${GPU_NODE}" != "true" ]]; then
+export -f should_skip_nvidia_drivers
+skip_nvidia_driver_install=$(retrycmd_if_failure_no_stats 10 1 10 bash -c should_skip_nvidia_drivers)
+ret=$?
+if [[ "$ret" != "0" ]]; then
```

Comment:
should this be terminal? I think yes if we call it a real contract.

Reply:
you mean having an explicit exit code for failure during this step?

Reply:
should I call the `exit` at all, or let CSE keep going but install the drivers by default

I think fail, if we'll have users relying on this

Reply:
yeah I think hard failure makes more sense